src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +102,1361 @@ static DamageTable D_80BBF858 = {
+        Matrix_RotateZS(this->bodyRotate.z, MTXMODE_APPLY);
+        Matrix_Push();
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBECBC.s")
+void EnSGoro_DrawUnrolled(EnSGoro* this, PlayState* play) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBEEB4.s")
+    SkelAnime* skelAnime;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBEF34.s")
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBEFA0.s")
+    gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(sEyeTextures[this->eyeTexIndex]));
+    gDPPipeSync(POLY_OPA_DISP++);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBF01C.s")
+    skelAnime = &this->skelAnime;
+    SkelAnime_DrawTransformFlexOpa(play, skelAnime->skeleton, skelAnime->jointTable, skelAnime->dListCount,
+                                   EnSGoro_OverrideLimbDraw, NULL, EnSGoro_TransformLimbDraw, &this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/EnSGoro_Init.s")
+    CLOSE_DISPS(play->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/EnSGoro_Destroy.s")
+void EnSGoro_DrawRolledUp(EnSGoro* this, PlayState* play) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/EnSGoro_Update.s")
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBF298.s")
+    Matrix_Translate(this->actor.world.pos.x, this->actor.world.pos.y + this->actor.shape.yOffset,
+                     this->actor.world.pos.z, MTXMODE_NEW);
+    Matrix_RotateYS(this->actor.shape.rot.y, MTXMODE_APPLY);
+    Matrix_Translate(0.0f, -this->actor.shape.yOffset, 0.0f, MTXMODE_APPLY);
+    Matrix_RotateZS(this->actor.shape.rot.z, MTXMODE_APPLY);
+    Matrix_Translate(0.0f, this->actor.shape.yOffset, 0.0f, MTXMODE_APPLY);
+    Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBF3D0.s")
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, gGoronRolledUpDL);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBF3EC.s")
+    CLOSE_DISPS(play->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBF5F0.s")
+void EnSGoro_Draw(Actor* thisx, PlayState* play) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBF6BC.s")
+    EnSGoro* this = (EnSGoro*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/EnSGoro_Draw.s")
+    if ((this->actionFlags & ENSGORO_ACTIONFLAG_ROLLEDUP)) {
+        EnSGoro_DrawRolledUp(this, play);
+        return;
+    }
+    EnSGoro_DrawUnrolled(this, play);
```

Comment:
```suggestion
        EnSGoro_DrawRolledUp(this, play);
    } else {
        EnSGoro_DrawUnrolled(this, play);
    }
```
Generally we try to avoid early returns when possible.